File: backend/app/schemas/hall_of_fame.py
"""
班级光荣榜响应 schema
"""
from typing import Optional
from pydantic import BaseModel


class ChampionItem(BaseModel):
    """单个上榜学生条目"""
    user_id: int
    nickname: str
    hero_id: Optional[str]
    metric: int  # 数值（次数 / 秒数 / 分数差）
    metric_label: str  # 中文展示，如 "12 次满分通关"


class HallOfFameResponse(BaseModel):
    """班级光荣榜响应（任意一项可能为 null）"""
    class_id: Optional[int]
    class_name: Optional[str]
    period: str  # 如 "2026-05"
    champions: dict

File: frontend/src/api/hallOfFame.ts
import client from './client';

export interface ChampionItem {
  user_id: number;
  nickname: string;
  hero_id: string | null;
  metric: number;
  metric_label: string;
}

export interface HallOfFameResponse {
  class_id: number | null;
  class_name: string | null;
  period: string;
  champions: {
    perfect_king: ChampionItem | null;
    speed_king: ChampionItem | null;
    progress_star: ChampionItem | null;
  };
}

export async function getClassHallOfFame(): Promise<HallOfFameResponse> {
  return client.get('/student/class/hall-of-fame');
}
